Gifski/MainWindowController.swift

```diff
@@ -0,0 +1,167 @@
+import Cocoa
+import ProgressKit
+import DockProgress
+
+class MainWindowController: NSWindowController {
+
+	private var progressObserver: NSKeyValueObservation?
+
+	lazy var circularProgress = with(CircularProgressView(frame: CGRect(widthHeight: 160))) {
+		$0.foreground = .appTheme
+		$0.strokeWidth = 2
+		$0.percentLabelLayer.setAutomaticContentsScale()
+		$0.percentLabelLayer.implicitAnimations = false
+		$0.layer?.backgroundColor = .clear
+		$0.isHidden = true
+		$0.centerInWindow(window)
+	}
+
+	lazy var videoDropView = with(VideoDropView()) {
+		$0.frame = window!.contentView!.frame
+		$0.dropText = "Drop a Video to Convert to GIF"
+		$0.onComplete = { url in
+			self.convert(url.first!)
+		}
+	}
+
+	var choosenDimensions: CGSize = .zero
+	var choosenFrameRate: Int?
+
+	var isRunning: Bool = false {
+		didSet {
+			if isRunning {
+				videoDropView.isHidden = true
+			} else {
+				videoDropView.fadeIn()
+			}
+
+			circularProgress.isHidden = !isRunning
+		}
+	}
+
+    override func windowDidLoad() {
+        super.windowDidLoad()
```

Comment:
Incorrect indentation.

Reply:
From the docs:

> The default implementation does nothing.

So calling `super` isn't strictly needed. What do you think about it? Should we still do it for consistency?

Reply:
I think we should remove it 👍